### PR TITLE
Adds a flag that determines if you can toggle auto unloading on a gun

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Stuff like disposables didn't clear their list of chambered refs, sad.
Also fixes unfunny exploit.
## Why It's Good For The Game

Bug bad.
## Changelog
:cl:
fix: Fixed one shot rocket launchers not clearing their chambered items after firing
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun_system.dm [hunk 1]
```diff
@@ -832,6 +832,8 @@
 		if(length(chamber_items) && CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_AUTO_EJECT) && CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_MAGAZINES) && get_current_rounds(chamber_items[current_chamber_position]) < (!CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_REQUIRES_UNIQUE_ACTION) ? rounds_per_shot : 0))
 			playsound(src, empty_sound, 25, 1)
 			unload(after_fire = TRUE)
+		if(length(chamber_items) && CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_CLEAR_CHAMBER_AFTER_SHOT))
+			QDEL_LIST(chamber_items)
```
Comment (Lumipharon) [inline code/modules/projectiles/gun_system.dm:836]: Why's it qdel'ing?

Wouldn't it just be better to have a flag or something that prevent auto eject toggling? Its not just the RL issue caused by that.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_system.dm (hunk 1 above)
Comment (ivanmixo) [reply code/modules/projectiles/gun_system.dm:836]: Ehhh the disposable RL keeps the rocket it fired in it's chamber so. Probably

**Review by Lumipharon — APPROVED** [review]
The special pistol needs the flag as well btw

File: code/modules/projectiles/gun_helpers.dm [hunk 2]
```diff
@@ -569,6 +569,10 @@ As sniper rifles have both and weapon mods can change them as well. ..() deals w
 	set name = "Toggle Automatic Magazine Ejection (Weapon)"
 	set desc = "Toggles the automatic unloading of the gun's magazine upon depletion."
 
+	if(CHECK_BITFIELD(reciever_flags, AMMO_RECIEVER_AUTO_EJECT_LOCKED))
+		balloon_alert(usr, "[src] doesn't support toggling of automatic ejection.")
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_helpers.dm:573]:
```suggestion
		balloon_alert(usr, "Cannot toggle ejection")
```

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**